tests/shardddl3/run.sh

```diff
@@ -871,6 +871,1199 @@ function DM_112() {
+        "alter table partition is unsupported" 1
+}
+
+# Add partitioning
+function DM_140 {
+    run_case 140 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 140 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_141_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10),(11),(12),(13),(14),(15);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(20),(21),(22),(23),(24),(25);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(30),(31),(32),(33),(34),(35);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} remove partitioning"
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status test" \
+        "ALTER TABLE \`${shardddl1}\`.\`${tb1}\` REMOVE PARTITIONING" 1 \
+        "Unsupported remove partitioning" 1
+}
+
+# Remove partitioning.
+function DM_141 {
+    run_case 141 "double-source-pessimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id)(partition p0 values less than (100));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id)(partition p0 values less than (100));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key) partition by range(id)(partition p0 values less than (100));\"" \
+        "clean_table" "pessimistic"
+    run_case 141 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id)(partition p0 values less than (100));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id)(partition p0 values less than (100));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key) partition by range(id)(partition p0 values less than (100));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_142_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10),(11),(12),(13),(14),(15);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(20),(21),(22),(23),(24),(25);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(30),(31),(32),(33),(34),(35);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add partition (partition p1 values less than (200));"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(110),(111),(112),(113),(114),(115);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add partition (partition p1 values less than (200));"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(120),(121),(122),(123),(124),(125);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add partition (partition p1 values less than (200));"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(130),(131),(132),(133),(134),(135);"
```

Comment:
ditto

Reply:
run_sql_source can guarantee the sql statement is executed successfully by MySQL. I think it's not our responsibility to guarantee the right  behavior of MySQL.